acquisitions/src/main/resources/thunderjet/mod-gobi/features/gobi-api-tests.feature

```diff
@@ -80,3 +80,272 @@ Feature: mod-gobi api tests
+    And headers { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitoken)', 'Accept': '*/*' }
+    When method GET
+    Then status 200
+    And match response.mappingType == 'Custom'
+    And match response.orderMappings.orderType == 'UnlistedPrintSerial'
+    And match response.orderMappings.mappings[1].dataSource.default == "false"
+    And match response.orderMappings.mappings[2].dataSource.default == "true"
+
+    # Update existing custom mapping
+    Given path '/gobi/orders/custom-mappings/UnlistedPrintSerial'
+    And headers { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitoken)', 'Accept': '*/*' }
+    And request valid_mapping
+    When method PUT
+    Then status 204
+
+    # Delete existing custom mapping
+    Given path '/gobi/orders/custom-mappings/UnlistedPrintSerial'
+    And headers { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitoken)', 'Accept': '*/*' }
+    When method DELETE
+    Then status 200
+
+    # Verify original mapping
+    Given path '/gobi/orders/custom-mappings/UnlistedPrintSerial'
+    And headers { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitoken)', 'Accept': '*/*' }
+    When method GET
+    Then status 200
+    And match response.mappingType == 'Default'
+    And match response.orderMappings.orderType == 'UnlistedPrintSerial'
+    And match response.orderMappings.mappings[1].dataSource.default == "true"
+    And match response.orderMappings.mappings[2].dataSource.default == "false"
+
+
+  Scenario: Make sure updated mapping is applied whenever placing orders
+    * def sample_po_original = read('classpath:samples/mod-gobi/po-unlisted-print-serial.xml')
+    * def sample_po_updated = read('classpath:samples/mod-gobi/po-unlisted-print-serial-updated.xml')
+
+    # Put an order for original mapping
+    Given path '/gobi/orders'
+    And headers { 'Content-Type': 'application/xml', 'x-okapi-token': '#(okapitoken)', 'Accept': '*/*' }
+    And request sample_po_original
+    When method POST
+    Then status 201
+    * def poLineNumber = /Response/PoLineNumber
+
+    # Check order approved
+    Given path '/orders/composite-orders'
+    And headers headers
+    And param query = 'poNumber==*' + poLineNumber.split('-')[0]+'*'
+    When method GET
+    Then status 200
+    And match response.purchaseOrders[0].approved == true
+
+    # Update mapping
+    * def valid_mapping = read('classpath:samples/mod-gobi/unlisted-print-serial.json')
+    Given path '/gobi/orders/custom-mappings'
+    And headers { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitoken)', 'Accept': '*/*' }
+    And request valid_mapping
+    When method POST
+    Then status 201
+
+    # Try to put an order for original mapping
+    Given path '/gobi/orders'
+    And headers { 'Content-Type': 'application/xml', 'x-okapi-token': '#(okapitoken)', 'Accept': '*/*' }
+    And request sample_po_original
+    When method POST
+    Then status 500
```

Comment:
**Question**
Why do we have 500 here? What errror exactly is returned from the server?

Reply:
As the mapping gets updated, a new field becomes required to be present in the po, which is not included in sample_po_original

Reply:
But what is the server error then? Should we return 4xx instead?
`return 4xx as the problem is with the data being sent and not the server program. 5xx are reserved for server errors`
(c) Google

Reply:
we can improve this by adding proper status code in scope of next story https://folio-org.atlassian.net/browse/MODGOBI-203
The current story is only for covering module with more karate tests, changing mod-gobi logic is out of scope of this story